xplique/concepts/craft_torch.py

```diff
@@ -0,0 +1,267 @@
+
+"""
+CRAFT Module for Pytorch
+"""
+
+from typing import Callable, Optional, Tuple
+from math import ceil
+import torch
+from torch import nn
+import numpy as np
+
+from .craft import BaseCraft, BaseCraftManager
+
+def _batch_inference(model: torch.nn.Module,
+                     dataset: torch.Tensor,
+                     batch_size: int = 128,
+                     resize: Optional[int] = None,
+                     device: Optional[str]='cuda') -> torch.Tensor:
+    """
+    Compute the model predictions of the input images.
+
+    Parameters
+    ----------
+        model
+            The model to use for inference.
+        dataset
+            The input images to be processed, of shape (n_samples, channels, height, width).
+        batch_size
+            The batch size to use during training and prediction. Defaults to 128.
+        resize
+            Optional argument to resize the outputs of the model. Defaults to None.
+        device
+            The compute device. Defaults to 'cuda'.
+
+    Returns
+    -------
+        activations
+            The latent activations of shape (n_samples, channels, height, width).
+    """
+    # dataset: (N, C, H, W)
+    nb_batchs = ceil(len(dataset) / batch_size)
+    start_ids = [i*batch_size for i in range(nb_batchs)]
```

Comment:
I don't understand why you do not simply batch the dataset and iterate over batches instead of using indexes ?

Reply:
It's because it's a numpy array/Tensor at this level, so I don't have a quick way to batch in a single line of code (unless I am mistaken :) )